lib/shared/errors.js

```diff
@@ -0,0 +1,366 @@
+/*
+ * Copyright 2016 resin.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+const _ = require('lodash');
+
+/**
+ * @summary Human-friendly error messages
+ * @namespace HUMAN_FRIENDLY
+ * @public
+ */
+exports.HUMAN_FRIENDLY = {
+
+  /* eslint-disable new-cap */
+
+  /**
+   * @namespace ENOENT
+   * @memberof HUMAN_FRIENDLY
+   */
+  ENOENT: {
+
+    /**
+     * @property {Function} title
+     * @memberof ENOENT
+     * @param {Error} error - error object
+     * @returns {String} message
+     */
+    title: (error) => {
+      return `No such file or directory: ${error.path}`;
+    },
+
+    /**
+     * @property {Function} description
+     * @memberof ENOENT
+     * @param {Error} error - error object
+     * @returns {String} message
+     */
+    description: _.constant('The file you\'re trying to access doesn\'t exist')
```

Comment:
Nice! Yeah, I was just thinking a class might have been optimal to re-use the hidden class in v8 (to avoid thrashing the heap with new objects), but that's probably over-optimising it ;)